src/dwarf.cpp

```diff
@@ -391,6 +294,124 @@ void line_header::read(freader& s, const file_details& details) {
 
 /**************************************************************************************************/
 
+attribute* alloc_attributes(std::size_t n) {
+#if ORC_FEATURE(LEAKY_MEMORY)
+    thread_local auto& pool_s = *(new std::list<std::unique_ptr<attribute[]>>);
+#else
+    thread_local std::list<std::unique_ptr<attribute[]>> pool_s;
+#endif // ORC_FEATURE(LEAKY_MEMORY)
+    constexpr auto block_size_k{32 * 1024 * 1024};
+    constexpr auto max_attributes_k{block_size_k / sizeof(attribute)};
+    thread_local std::size_t n_s{max_attributes_k};
+    if ((n_s + n) >= max_attributes_k) {
+        pool_s.push_back(std::make_unique<attribute[]>(max_attributes_k));
+        n_s = 0;
+    }
+    attribute* result = &pool_s.back().get()[n_s];
+    n_s += n;
+    return result;
+}
```

Comment:
This is the new routine that allocates 32MB blocks worth of `attribute` instances, and doles them out as necessary to `die`s that need them. Note the leaky memory switch, so they are *not* cleaned up at app shutdown when that switch is on.